### PR TITLE
Redesign 2: Add new functions and optimize code

### PR DESCRIPTION
**Pull request by azeemshatp**
What does this PR do?
- Add new functions and code optimization.
- Split the functions attach_gprs() and enable_bearer().
- Bluetooth functionality has previously been added through a sub-branch.

What are the relevant tickets? 
NA

Where should the reviewer start?
SIM800-AT.cpp

How should this be manually tested?
Flash the CS code and see if all the states pass successfully

### CODE REVIEW

File: src/SIM800-AT.cpp [hunk 1]
```diff
@@ -323,40 +350,54 @@ int GPRS::get_ip(char *resp_buf, int size_buf)
...
+{
+    // Close the GPRS PDP context.
+    send_cmd("AT+CIPSHUT");
+}
+
 int GPRS::close_tcp(char *resp_buf, int size_buf)
 {
     // closes the TCP connection
     send_cmd("AT+CIPCLOSE"); // Return "CLOSE OK"
     read_resp(resp_buf, size_buf, DEFAULT_TIMEOUT, NULL);
```
Comment (JordanSilverman) [inline src/SIM800-AT.cpp:382]: Should we still check for "CLOSE OK" here? Could increase the timeout a bit.

File: src/SIM800-AT.cpp [hunk 2]
```diff
@@ -371,7 +412,7 @@ int GPRS::detach_gprs(char *resp_buf, int size_buf)
 int GPRS::disable_bearer(char *resp_buf, int size_buf)
 {
     send_cmd("AT+SAPBR=0,1");
-    if (0 != read_resp(resp_buf, size_buf, DEFAULT_TIMEOUT, "OK")) {
+    if (0 != read_resp(resp_buf, size_buf, 10, "OK")) {
```
Comment (JordanSilverman) [inline src/SIM800-AT.cpp:415]: Reduce the timeout here otherwise it will trigger the watchdog

File: src/SIM800-AT.cpp (hunk 1 above)
Comment (azeemshatp) [reply src/SIM800-AT.cpp]: That was missed actually. Updated now.

File: src/SIM800-AT.cpp (hunk 2 above)
Comment (azeemshatp) [reply src/SIM800-AT.cpp]: Reduced to 6 now

**Pull request merged**